### DISCUSSION TITLE
remove a specific marker

### DISCUSSION THREAD

**Discussion by alisafa1371** [Leaflet/Leaflet#8439] [Q&A] [answered]
Hi guys, could you please help me how can I remove a specific marker from leaflet map?

**Comment by Falke-Design** [comment, marked as answer]
`map.removeLayer(YOUR_MARKER_OBJECT)`

**Reply by alisafa1371** [reply]
Thank you. and how can I remove it with coordinates or Id? the only thing that I have from different Markers is it's coordinates or _icon.id or _leaflet_id

**Reply by alisafa1371** [reply]
Thank you my friend I Solved it ! I wanted to delete a specific marker when I do something (for example click delete button) out side of the map and I wanted to delete the exact marker which is some how related to that marker . so at the creating marker stage I gave the marker an Id (equal to id of the element outside the map) using Marker-object._icon_leaflet_id and also push it to an array and at deleting stage I use find method to find the marker that has the id that I want and finally I use map.removeLayer(SELECTED_MARKER_OBJECT)

**Reply by Jesusson-softwareandIT** [reply]
What is in the SELECTED MARKER OBJECT, is it id or what
